yesod-auth/Yesod/Auth.hs

```diff
@@ -421,14 +421,20 @@ authLayoutJson w json = selectRep $ do
 --
 -- @since 1.1.7
 clearCreds :: (MonadHandler m, YesodAuth (HandlerSite m))
-           => Bool -- ^ if HTTP redirect to 'logoutDest' should be done
+           => Bool -- ^ if HTTP redirect to 'logoutDest', if JSON respond with message.
```

Comment:
It seems a bit strange to me that we conditionally respond with a JSON body like this. What if we _always_ respond with a JSON body if the client accepts JSON responses?

Reply:
That's a good idea. Will update.